addons/hr_attendance/models/hr_attendance.py

```diff
@@ -278,6 +279,8 @@ def create(self, vals_list):
         return res
 
     def write(self, vals):
+        if vals.get('employee_id') and vals['employee_id'] != self.env.user.employee_id.id and not self.env.user.has_group('hr_attendance.group_hr_attendance_user'):
```

Comment:
In case the user is logged to multiple companies and they are editing the attendances of their own employee from a company other than the primary one :-)

```suggestion
        if vals.get('employee_id') and vals['employee_id'] not self.env.user.employee_ids.ids and not in self.env.user.has_group('hr_attendance.group_hr_attendance_user'):
```